### DISCUSSION TITLE
Should query params be built into the url or use a query builder inside fetcher and use the tuple composition?

### DISCUSSION THREAD

**Discussion by lucksp** [vercel/swr#1328] [Q&A]
Many of our get endpoints require on query params.  Sometimes we just build the string as URL.  Also, we have a fetcher function that takes in an optional object to build the string and generate the updated URL string which is used by fetch.

I recently noticed SWR has the tuple pattern as the first argument to the url: https://swr.vercel.app/docs/arguments#multiple-arguments

Does SWR have any recommended approaches to query param url strings?  Should the url just be built ahead of time, and therefore the params change as part of url string, or is it preferred to have the static URL and pass in values to the tuple?

This:
```
const url = /some_endpoint?query=true&param=false;
const {data} = useSWR(url, fetcher);
```

vs

```
const url = /some_endpoint;
const params = {
  query: true,
  param: false,
}
const {data} = useSWR([url, params], (swrUrl) => fetcher(swrUrl, {
     termBuilder:  { ...params }
  }
);
```

**Comment by shuding** [comment]
Both are fine I think. If you don’t need to access any specific parameter in `fetcher`, you can go with the first approach.
